### PR TITLE
Auto Rifle Sprites and Bug Fixes

### LINKED ISSUE #11818: Sec auto rifles still have fire modes
secrifles

the PR specified that it no longer had firing modes, and indeed they don't act differently, but the framework for firing modes is still active on the guns.

### PR DESCRIPTION
**Pull request by Steelpoint**
Fixes https://github.com/tgstation/-tg-station/issues/11818

---

New security auto rifle sprites, NT auto guns get the blue sprite while a 'new' gun get the more syndicate looking red spriets.

screenshot 2015-09-17 23 26 20

---
- Security Auto Rifle
  Sprite changed, has a blue appearance. Mechanics unchanged.
- Hacked Auto Rifle (NOT IN THIS PR)
  New weapon RnD can produce. Requires syndicate research to be at level three. Is cheaper to produce than a SABR but is obviously less effective.

In addition the hacked auto rifle has a burst fire option (2 rounds per burst) and is locked into auto firing, you cannot switch to single shot.

The rifle uses the above red sprites. RnD produced guns do not have any firing pins loaded.

(Both Auto Rifles now emit the empty gun sound the C20r uses when the magazine is empty)
- SABR (NOT IN THIS PR)
  Nerfed to hold only 30 bullets per magazine.
- Projectiles
  Projectile bullets changed to a far more clear and easier to distinguish sprite.

---

Miscellaneous bug fixes, some already in the following PR: (https://github.com/tgstation/-tg-station/pull/11849)

---

### CODE REVIEW

**Comment by Steelpoint** [comment]
I might remove the gun indicators.

**Comment by Steelpoint** [comment]
Ok no more indicators, its just the nice looking SMG.

screenshot 2015-09-17 15 27 17

**Comment by Dorsisdwarf** [comment]
Can you experiment with giving it an NT-blue case instead of syndicate black-all-over?

**Comment by Steelpoint** [comment]
You mean sort of like the HoS gun?

**Comment by Ausops** [comment]
what do you mean NT-blue, do you mean like the shade that exofabs or other machines have

when you say blue the only thing that comes to mind is the blue that non-lethal weapons in MGSV have

**Comment by Steelpoint** [comment]
Well here's a blue version that's akin to the style the HoS gun uses.

screenshot 2015-09-17 16 20 00

**Comment by Dorsisdwarf** [comment]
I meant a mix of that and the current one, so it has that gunmetal-blue case, but the grip and barrel are still black.

The only way I can think to explain what I mean is "google lasgun". That sort of shell.

**Comment by Steelpoint** [comment]
I'll work on a sprite then.

**Comment by Steelpoint** [comment]
Something like this?

screenshot 2015-09-17 20 31 02

**Comment by WJohn** [comment]
:+1: to the sprites. If these are removed, we should at least keep them in the code for admins to spawn.

**Comment by Steelpoint** [comment]
I'll leave the current gun sprites in then.

Also WJ did you mean like the fully grey one or the blue cased one above?

**Comment by Dorsisdwarf** [comment]
That's perfect, Steelpoint. Just enough blue to look like something NT might use, not enough blue to look cartoony.
of course I hate these guns still but now at least they look classy.

**Comment by WJohn** [comment]
I think I like the blue one slightly more for NT use, but the black is nice, too.

**Comment by Steelpoint** [comment]
I'll use the above blue one.

If I ever go forward with the free syndi SMG I might use a red tinted SMG from above. But that's offtopic.

**Comment by Steelpoint** [comment]
Ok I'm going to fix a bunch of other minor issues, so watch this space.

**Comment by Steelpoint** [comment]
In game image of new sprites.

Along with a alt red one, likely to be used by a illegal weapon version in RnD.

screenshot 2015-09-17 23 26 20

**Comment by bustygoku** [comment]
I was gonna be all 'eww syndicate weapons for nanotrasen' but that blue color scheme two-tone is sexy as fuck.

**Comment by Steelpoint** [comment]
Updated OP with new changes.

**Comment by Anonus** [comment]
Not even NT blue/10
Still not energy projectiles/10

**Comment by Lumipharon** [comment]
blue sprite looks nice.
Red sprite is poo.

Rifles still should not exist round start.
Also can you please show these new bullet sprites in action? I hope they're not those sprites from when someone tried to change how shooting worked.

**Comment by Dorsisdwarf** [comment]
Needs pic of booletsprite

**Comment by Steelpoint** [comment]
Here's a older gif of the sprites from Goof's pixel PR

https://camo.githubusercontent.com/e50cef441f0fe7b35c8f606574ed03fbcc9155e8/687474703a2f2f692e696d6775722e636f6d2f797353566f774f2e676966

**Comment by Lumipharon** [comment]
Wheeelp. It's the ugly sprites.

File: code/modules/research/designs/weapon_designs.dm [hunk 1]
```diff
@@ -221,24 +231,24 @@
...
 	build_path = /obj/item/ammo_box/magazine/smgm9mm/fire
 
 /datum/design/mag_smg/incin_tox
-	name = "Submachine Gun Urnaium Magazine (9mmIC)"
-	desc = "A 42-round urnaium tipped round magazine for the prototype submachine gun. Deals toxin damage, but less overall damage."
+	name = "SABR SMG Urnaium Magazine (9mmTX)"
+	desc = "A 30-round urnaium tipped round magazine for the prototype submachine gun. Deals toxin damage, but less overall damage."
 	id = "mag_smg_tx"
-	materials = list(MAT_METAL = 3000, MAT_GLASS = 200, MAG_URNAIUM = 1000)
+	materials = list(MAT_METAL = 3000, MAT_GLASS = 200, MAT_URNAIUM = 1000)
```
Comment (GunHog) [inline code/modules/research/designs/weapon_designs.dm]: Do not forget to change it to "URANIUM". Please remember to test and verify each portion of your PR in order to catch such bugs.

File: code/modules/research/designs/weapon_designs.dm [hunk 2]
```diff
@@ -188,31 +198,31 @@
...
+	materials = list(MAT_METAL = 6000, MAT_SILVER = 200, MAT_GLASS = 800)
 	build_path = /obj/item/ammo_box/magazine/wt550m9/wtic
 
 /datum/design/mag_oldsmg/tx_mag
-	name = "Submachine Gun Urnaium Magazine (9mmTX)"
+	name = "WT-550 Auto Gun Urnaium Magazine (9mmTX)"
 	desc = "A 20 round urnaium tipped magazine for the out of date security WT-550 Auto Rifle"
 	id = "mag_oldsmg_tx"
-	materials = list(MAT_METAL = 6000, MAG_SILVER = 400, MAG_URNAIUM = 2000)
+	materials = list(MAT_METAL = 6000, MAT_SILVER = 400, MAT_URNAIUM = 2000)
```
Comment (GunHog) [inline code/modules/research/designs/weapon_designs.dm]: Change to "URANIUM" here, as well.

File: code/modules/projectiles/ammunition/magazines.dm [hunk 3]
```diff
@@ -173,7 +173,7 @@
 	icon_state = "smg9mm-42"
 	ammo_type = /obj/item/ammo_casing/c9mm
 	caliber = "9mm"
-	max_ammo = 42
+	max_ammo = 30
```
Comment (phil235) [inline code/modules/projectiles/ammunition/magazines.dm]: since you change the max ammo to 30 you might want to rename the icon_state to "smg9mm-30"

Comment (Steelpoint) [inline code/modules/projectiles/ammunition/magazines.dm]: I did, don't worry.

**Comment by Steelpoint** [comment]
This is good to merge.

**Comment by Aranclanos** [comment]
i love how you put a pr with bugfixes with new shit so it can't be merged without a shitstorm unless its stealth

**Comment by Steelpoint** [comment]
I did state all the changes in the OP though?

**Comment by Steelpoint** [comment]
I mean if you want me to separate the changes to the bug fixes that's fine, but the only (noticeable) changes here are a sprite change and a small nerf to the SABR.

**Comment by GunHog** [comment]
I am pretty sure he is talking about the new hacked rifle as well. What started as a bug fix to the Sec rifle and a new sprite for it is now a PR with a new item and a balance change to an existing one. 

It is recommended that you put the SABR mag nerfs and hacked auto rifle in their own PR, so no one has to take the extra time to bother with balance changes.

**Comment by Steelpoint** [comment]
Ok I removed the hacked rifle and ammo changes.

**Pull request merged**